### DISCUSSION TITLE
Serialize Relationships as null

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#2041] [Ideas]
It would be nice to be able to serialize relationships as null since for some use cases, you want to have a computed property of multiple relationships and serialize them as null to hide them from responses, I tried using undefined but they are still showing, this code although works it's throwing a type error:

```
    //@ts-ignore
    serializeAs: null,
```

**Comment by steschwa** [comment]
I agree, that would be useful.
Anyway you can manually serialize the model using `model.serialize()` and hide the relationship with
` relationShipProperty: undefined `. This way they are not serialized to JSON.

Like so:

**In a controller:**
```ts
public async method({ params }: HttpContextContract) {
   const user = User.findOrFail(params.id)

   // Business logic

  return {
    ...user.serialize(),
   posts: undefined,
   postsCount: user.posts.length
  }
}
```

Will yield the response:
```json
{
   "id": 1,
   .... rest user props
  "posts_count": 5
}
```

**Reply by thetutlage** [reply]
You can hide relationships by simply setting the `serializeAs = null`, just like you would do in the `@column` decorator.

```ts
  @hasMany(() => Post, { serializeAs: null })
  public tokens: HasMany<typeof Post>
```

**Reply by popovicieduard** [reply]
Forgot to mention that the relationship is many to many, it seems that hasMany has no problem with null value but manytomany has:
https://prnt.sc/w281zi

**Reply by thetutlage** [reply]
It will be nice if you share proper code. One liners doesn't help much

**Reply by popovicieduard** [reply]
```
  @manyToMany(() => Company, {
    localKey: 'id',
    pivotForeignKey: 'foobar',
    relatedKey: 'id',
    pivotRelatedForeignKey: 'foobar',
    pivotTable: 'foobar',
    pivotColumns: ['foobar'],
    serializeAs: null, //type error
  })
  public fooBar: ManyToMany<typeof Company>
```

**Reply by popovicieduard** [reply]
The above code works if I add a //@ts-ignore to it, so I think only relations.d.ts has to be modified.
```
    type ManyToManyRelationOptions<Related extends ModelRelations> = {
        pivotTable?: string;
        localKey?: string;
        pivotForeignKey?: string;
        relatedKey?: string;
        pivotRelatedForeignKey?: string;
        pivotColumns?: string[];
        serializeAs?: string; // -> becomes serializeAs?: string | null
        onQuery?(query: Related['builder'] | Related['subQuery']): void;
    };
```

**Reply by thetutlage** [reply]
Yup, this needs to be fixed. Happy to accept the PR if you can open one.

**Reply by popovicieduard** [reply]
sure
